### DISCUSSION TITLE
Action times out on Laravel Vapor

### DISCUSSION THREAD

**Discussion by madsem** [filamentphp/filament#3174] [Help]
I am facing a curious problem, in my local docker environment that resembles Vapor and uses the same Mysql 8 version everything works.

When deploying to production on Laravel Vapor, a particular action times out.
Initially I thought it's my code, or Vapor config. So I increased memory in `vapor.yml` as well as timeout to 20 seconds.

But it still happens, the action performs a minimal action (replicating a model):
```php
Action::make('clone')
                ->label('Use Template')
                ->color('secondary')
                ->modalButton('Create Your Lander')
                ->modalHeading('Clone This Template Into Your Landers')
                ->icon('heroicon-s-duplicate')
                ->requiresConfirmation()
                ->action(function (Model $record) {

                    $clone = collect($record->toArray())->except([
                        'identifier',
                        'user_id',
                        'type',
                        'name',
                    ]);

                    $clone->put('name', 'Cloned Lander: Give me a descriptive name');

                    $lander = new LandingPage($clone->toArray());
                    $cloned = \Auth::user()->landingPages()->save($lander);

                    redirect(route('landers.edit', $cloned));
                }),
```

The above times out, after 20 seconds... Without having replicated the model, so it never reaches the `redirect`.

Any ideas if something in Filament could cause this? 

PS: What happens exactly is:

1. Click action button
2. Confirm ConfirmationModal
3. Waiting 20 seconds
4. Error Modal pops up, completely black with: `500 server error` & no replicated model in DB.

No exceptions in Telescope, other than  a Cloudwatch log entry:
`
2022-07-19T15:36:22.748Z 4ae75d89-e5a2-4b4b-b6c3-4b7819b0d4f3 Task timed out after 20.02 seconds`

**Comment by danharrin** [comment]
Remove parts of the action and try again to work out what line causes the timeout

**Reply by madsem** [reply]
Thanks @danharrin 

I'll try to add some logging tomorrow and make some deployments. I wonder why this works flawlessly local &trade;
